solana/transaction.py

```diff
@@ -132,7 +132,7 @@ def compile_message(self) -> Message:
 
         account_metas, program_ids = [], set()
         for instr in self.instructions:
-            if not instr.program_id or not instr.keys:
+            if not instr.program_id:
                 raise AttributeError("invalid instruction:", instr)
             account_metas.extend(instr.keys)
```

Comment:
@Lucas-Bruder so for line 137? What are we suppose to push into `account_metas`?

Reply:
Do you need to update the default case here as well?
https://github.com/michaelhly/solana-py/blob/51570e0d5d8951dcb3d06402c8a40e59be546f9b/solana/transaction.py#L41

Reply:
im assuming it should be an empty list in this particular situation, other situations instr.keys will be populated so it'll fill account_metas. it is kinda a weird use case, but it seems like it's valid?

im learning the programming model as i go and have been heavily relying on solana-web3.js and anchor for reference. looks like between your library and solana-web3.js, this method is pretty much the same with the exception of this for loop.

for context, logging tx [here](https://github.com/project-serum/anchor/blob/master/ts/src/provider.ts#L193) for program.rpc.initialize() in example [basic-0](https://github.com/project-serum/anchor/blob/master/examples/tutorial/basic-0/tests/basic-0.js#L13) and it seems to work as expected without keys in that instruction.
```
Transaction {
  signatures: [
    {
      signature: <Buffer 8c 87 af e7 b5 4a 96 51 5a 8d 56 79 a4 56 fb be ef 40 55 fe 41 eb 09 a3 cf 3e 44 9d 6d 91 81 6e e5 62 d8 88 b0 cd aa bd 26 c4 77 cb 76 1d 95 2a 69 db ... 14 more bytes>,
      publicKey: [PublicKey]
    }
  ],
  instructions: [
    TransactionInstruction {
      data: <Buffer af af 6d 1f 0d 98 9b ed>,
      programId: [PublicKey],
      keys: []
    }
  ],
  feePayer: PublicKey {
    _bn: <BN: 2509213b6d7091bd41e221b0b838901bc79b6f84f0715ce377912c1d15e667a>
  },
  recentBlockhash: '7Tf8fJLkd4tLtndzfsyssVc1MbE1FqAgzWrnYSgzubP8'
}
```

Reply:
Okay. I released a new version with your patch.

Reply:
Actually whoops. I was asking if we should update the default with
```python3
keys: List[AccountMeta]  = []
```

Reply:
oh yeah, i guess it depends on the client. my client will just put an empty list there. up to you!